app/src/main/java/com/example/bigowlapp/utils/SupervisorSmsListener.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bigowlapp.utils;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.telephony.SmsMessage;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Schedule;
+import com.google.firebase.Timestamp;
+import com.google.i18n.phonenumbers.NumberParseException;
+
+import java.util.Calendar;
+
+public class SupervisorSmsListener extends BroadcastReceiver {
+
+    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
+    private static final int THIRTY_MINUTES = 1800;
+    private String scheduleId;
+
+    IsSmsSenderACurrentEventSupervisor isSmsSenderACurrentEventSupervisor = new IsSmsSenderACurrentEventSupervisor();
+
+    public void onReceive(Context context, Intent receiverIntent) {
+        if (receiverIntent.getAction().equals(SMS_RECEIVED)) {
+            Bundle bundle = receiverIntent.getExtras();
+            if (bundle != null) {
+                // get sms objects
+                Object[] pdus = (Object[]) bundle.get("pdus");
+                if (pdus.length == 0) {
+                    return;
+                }
+                // large message might be broken into many
+                SmsMessage[] messages = new SmsMessage[pdus.length];
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < pdus.length; i++) {
+                    messages[i] = SmsMessage.createFromPdu((byte[]) pdus[i]);
+                    sb.append(messages[i].getMessageBody());
+                }
+                String smsSenderNum = messages[0].getOriginatingAddress();
+                scheduleId = sb.toString();
```

Comment:
This section is a little confusing and vague and the comments are really vague and don't actually describe what seems to be going on.

Consider extracting this into a method or multiple methods with more descriptive statements [ex `getScheduleIdFromSms()`] and removing the comments

Reply:
I don't think the code with the present comments is hard to understand, but I can make it more understandable